src/Qir/qir-utils.ps1

```diff
@@ -34,163 +34,41 @@ function Build-CMakeProject {
-            #       -Xlinker /guard:cf -shared -o lib\QIR\Microsoft.Quantum.Qir.Runtime.dll  -Xlinker /implib:lib\QIR\Microsoft.Quantum.Qir.Runtime.lib \
-            #       -Xlinker /pdb:lib\QIR\Microsoft.Quantum.Qir.Runtime.pdb -Xlinker /version:0.0 lib/QIR/bridge-rt.obj \
-            #       lib/QIR/CMakeFiles/qir-rt-support-obj.dir/QirRange.cpp.obj lib/QIR/CMakeFiles/qir-rt-support-obj.dir/OutputStream.cpp.obj ....\
-            #       -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 -loldnames && cd ."
-            #   lld-link: error: duplicate symbol: malloc
-            #   >>> defined at C:\src\llvm_package_6923b0a7\llvm-project\compiler-rt\lib\asan\asan_win_dll_thunk.cpp:34
-            #   >>>            clang_rt.asan_dll_thunk-x86_64.lib(asan_win_dll_thunk.cpp.obj)
-            #   >>> defined at ucrtbased.dll
-            #   clang++: error: linker command failed with exit code 1 (use -v to see invocation)
-            $sanitizeFlags += " -fsanitize=address"   # https://clang.llvm.org/docs/AddressSanitizer.html
-            #   TODO:
-            #       * Some tests verify the failure behavior, i.e. they cause `Fail()` to be called and return to the caller with the exception. 
-            #         Any allocations made between the call and the exception throw (caught by `REQUIRE_THROWS()`) are leaking.
-            #         Extract such tests to a separate .cpp file or executable and compile with leak check off (or suppress leaks in that .cpp or executable only).
-
+        if (-not ($IsWindows)) {
             # Common for all sanitizers:
-            $sanitizeFlags += " -fsanitize-blacklist="      # https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#suppressing-errors-in-recompiled-code-ignorelist
-                                                            # https://releases.llvm.org/11.0.1/tools/clang/docs/SanitizerSpecialCaseList.html
+            $sanitizeFlags = " -fsanitize-blacklist="      # https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#suppressing-errors-in-recompiled-code-ignorelist
+            # https://releases.llvm.org/11.0.1/tools/clang/docs/SanitizerSpecialCaseList.html
             $sanitizeFlags += (Join-Path $Path .. UBSan.ignore)
-
-            $sanitizeFlags += " -fno-omit-frame-pointer"            # https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
-            $sanitizeFlags += " -fno-optimize-sibling-calls"        # https://clang.llvm.org/docs/AddressSanitizer.html
+            $env:CFLAGS += $sanitizeFlags
+            $env:CXXFLAGS += $sanitizeFlags
         } # if (-not ($IsWindows))
 
-        $env:CFLAGS   += $sanitizeFlags
-        $env:CXXFLAGS += $sanitizeFlags
     } # if ($Env:BUILD_CONFIGURATION -eq "Debug") 
 
 
-    if (($IsMacOS) -or ((Test-Path Env:/AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin"))))
-    {
+    if (($IsMacOS) -or ((Test-Path Env:/AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin")))) {
```

Comment:
I would recommend to do the unrelated changes like this one in a separate small preparatory PR. 

The current PR has certain logic relocation. And it takes some time to double-check that the correctness of the logic is preserved. The multiple small unrelated changes like this one complicate the analysis of the correctness.